src/test/java/hudson/plugins/ec2/util/AmazonEC2FactoryMockImpl.java

```diff
@@ -0,0 +1,66 @@
+package hudson.plugins.ec2.util;
+
+import java.net.URL;
+import java.util.Collections;
+
+import javax.annotation.Nullable;
+
+import org.mockito.Mockito;
+import org.mockito.stubbing.Answer;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.ec2.AmazonEC2;
+import com.amazonaws.services.ec2.model.DescribeRegionsResult;
+import com.amazonaws.services.ec2.model.Region;
+
+import hudson.plugins.ec2.EC2Cloud;
+
+public class AmazonEC2FactoryMockImpl extends AmazonEC2Factory {
```

Comment:
Then, once you've adapted the production Guice module class, you can specify a test version here to override the `AmazonEC2` instance returned.